traffic_ops/app/lib/MojoPlugins/DnssecKeys.pm

```diff
@@ -86,12 +86,9 @@ sub register {
 			#add to keys hash
 			$keys->{$key} = { zsk => [@zsk], ksk => [@ksk] };
 
-			#delivery services
-			#first get profile_id
-			my $profile_id = $self->get_profile_id_by_cdn($key);
-
-			#then get deliveryservices
-			my %search = ( profile => $profile_id );
+			#find the cdn's delivery services to generate keys for
+			my $cdn = $self->db->resultset('Cdn')->find( { name => $key } );
+			my %search = ( cdn_id => $cdn->id );
 			my @ds_rs = $self->db->resultset('Deliveryservice')->search( \%search , { prefetch => [ { 'cdn' => undef }]});
```

Comment:
couldn't you just define the CDN in the prefetch?  Right now it is undef

Reply:
new commit coming